test/TestProjects/ModelReaderWriterValidation-TypeSpec/src/Generated/Models/UnknownBaseModel.Serialization.cs

```diff
@@ -61,7 +61,7 @@ BaseModel IJsonModel<BaseModel>.Create(ref Utf8JsonReader reader, ModelReaderWri
             }
 
             using JsonDocument document = JsonDocument.ParseValue(ref reader);
-            return DeserializeUnknownBaseModel(document.RootElement, options);
+            return DeserializeBaseModel(document.RootElement, options);
```

Comment:
Here we fix the issue #4220

Reply:
calling this method instead of changing the implementation of `DeserializeUnknownBaseModel` because of the following reasons:
1. We will have a pattern that if we have `T IJsonModel<T>.Create(ref Utf8JsonReader reader, ModelReaderWriterOptions options)`, we call `DeserializeT`.
2. `DeserializeModel` should have single responsibility: deserialize the json element and return an instance of myself, therefore `DeserializeUnknownBaseModel` should only return an instance of `UnknownBaseModel`

Reply:
This still doesn't quite match what we did in System.ClientModel.

There we had IJsonModel.Create from the unknown type call DeserializeUnknown.  This method always forwards to DeserializeBase.

There is no implementation for DeserializeUnknown because you always need the logic on the base to run to pick which type to create.

Is there a reason we are leaving this implementation here?  If its because the default case on the DeserializeBase is supposed to call this that seems very strange that we are bouncing back and forth between the two classes.

Reply:
Lets open a tracking issue to discuss further.

Reply:
> Lets open a tracking issue to discuss further.

created: https://github.com/Azure/autorest.csharp/issues/4254